FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.DataClient/Api/ApiDataSource.cs

```diff
@@ -0,0 +1,58 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using EnsureThat;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Fhir.Synapse.Common;
+using Microsoft.Health.Fhir.Synapse.Common.Configurations;
+using Microsoft.Health.Fhir.Synapse.DataClient.Api.Dicom;
+using Microsoft.Health.Fhir.Synapse.DataClient.Exceptions;
+
+namespace Microsoft.Health.Fhir.Synapse.DataClient.Api
+{
+    /// <summary>
+    /// The data source from service search API.
+    /// </summary>
+    public class ApiDataSource : IApiDataSource
```

Comment:
would it be better to create two classes `FhirApiDataSource` and `DicomApiDataSource` derived from `IApiDataSource`？

Reply:
Currently the attributes of two data sources are the same. Maybe we can keep it as a common class until some differences are made.